Etabs_Adapter/CRUD/Create/Load.cs

```diff
@@ -273,32 +273,85 @@ public void SetLoad(BarPointLoad barPointLoad, bool replace)
+                double val = areaTemperatureLoad.TemperatureChange;
                 if (val != 0)
                     ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 1, val, "", replace);
             }
         }
 
         /***************************************************/
 
-        public void SetLoad(BarUniformTemperatureLoad barTempratureLoad, bool replace)
+        public void SetLoad(AreaDifferentialTemperatureLoad areaDifferentialTemperatureLoad, bool replace)
         {
             int ret = 0;
-            string caseName = GetAdapterId<string>(barTempratureLoad.Loadcase);
-            foreach (Bar bar in barTempratureLoad.Objects.Elements)
+            string caseName = GetAdapterId<string>(areaDifferentialTemperatureLoad.Loadcase);
+            foreach (IAreaElement area in areaDifferentialTemperatureLoad.Objects.Elements)
             {
-                double val = barTempratureLoad.TemperatureChange;
+                Dictionary<double, double> tempProfile = areaDifferentialTemperatureLoad.TemperatureProfile;
+                double tempMed = tempProfile.Average(kv => kv.Value);
+                double tempDelta= ((tempProfile[0] - tempMed) - (tempProfile[1] - tempMed));
+
+                if (tempMed!=0)
+                { ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 1, tempMed, "", replace);}
+
+                if (tempDelta!=0)
+                { ret = m_model.AreaObj.SetLoadTemperature(GetAdapterId<string>(area), caseName, 3, tempDelta, "", replace);}
+
+            }
+        }
+
+        /***************************************************/
+
+        public void SetLoad(BarUniformTemperatureLoad barTemperatureLoad, bool replace)
+        {
+            int ret = 0;
+            string caseName = GetAdapterId<string>(barTemperatureLoad.Loadcase);
+            foreach (Bar bar in barTemperatureLoad.Objects.Elements)
+            {
+                double val = barTemperatureLoad.TemperatureChange;
                 if (val != 0)
                     ret = m_model.FrameObj.SetLoadTemperature(GetAdapterId<string>(bar), caseName, 1, val, "", replace);
             }
         }
 
+        /***************************************************/
+
+        public void SetLoad(BarDifferentialTemperatureLoad barDifferentialTemperatureLoad, bool replace)
+        {
+            int ret = 0;
+            string caseName = GetAdapterId<string>(barDifferentialTemperatureLoad.Loadcase);
+            DifferentialTemperatureLoadDirection loadDir= barDifferentialTemperatureLoad.LoadDirection;
+
+            foreach (Bar bar in barDifferentialTemperatureLoad.Objects.Elements)
+            {
+                Dictionary<double, double> tempProfile = barDifferentialTemperatureLoad.TemperatureProfile;
+                double tempMed = tempProfile.Average(kv => kv.Value);
+                double tempDelta = ((tempProfile[0] - tempMed) - (tempProfile[1] - tempMed));
```

Comment:
As above, the BHoM representation allows nonlinear temperature profiles which ETABS does not. So we should raise a warning here to let the user know we will be taking the average and calculating the delta for top and bottom.

Also, as above - the Average is not adequate - assuming this is applied at the centreline of the `Bar`.

Reply:
@peterjamesnugent ,
thanks for your comment.
I've added a warning message for the user in the latest commit.
https://github.com/BHoM/ETABS_Toolkit/commit/7e99a88735805450663e7d454ddf50519748bd80

Reply:
Same comment as above, just blanket raising a warning here isn't enough

Reply:
As discussed offline, happy that the uneven temperature gradients are accounted for using the `tempDelta` to offset the temperature graidnet (`tempMed`). We can remove the warning about uneven temperature distributions, and we can flag a not implemented error for any nonlinear temperature profiles, stating Etabs does not support it.

Reply:
@peterjamesnugent,
updated warning message for non-linear profiles only incorporated in latest commit 4fbc139d1503e2901fa278146013f75642910ad4